src/pilot/dell_nfv.py

```diff
@@ -0,0 +1,443 @@
+                for line in rc_file:
+                    if 'OS_USERNAME=' in line:
+                        UC_USERNAME = line.split('OS_USERNAME=')[
+                            1].strip("\n").strip("'")
+                    elif 'OS_PASSWORD=' in line:
+                        cmd = 'sudo hiera admin_password'
+                        proc = subprocess.Popen(
+                            cmd,
+                            shell=True,
+                            stdin=subprocess.PIPE,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE)
+                        output = proc.communicate()
+                        output = output[0]
+                        UC_PASSWORD = output[:-1]
+                    elif 'OS_TENANT_NAME=' in line:
+                        UC_PROJECT_ID = line.split('OS_TENANT_NAME=')[
+                            1].strip("\n").strip("'")
+                    elif 'OS_AUTH_URL=' in line:
+                        UC_AUTH_URL = line.split('OS_AUTH_URL=')[
+                            1].strip("\n").strip("'")
+        except Exception as error:
+            message = "Exception {}: {}".format(
+                type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception(
+                "Failed to get undercloud details from"
+                " the undercloud rc file {}".format(file_path))
+
+    @classmethod
+    def get_overcloud_details(cls, overcloud_name):
+        try:
+            global UC_USERNAME, UC_PASSWORD, UC_PROJECT_ID, UC_AUTH_URL
+            file_path = home_dir + '/' + overcloud_name + "rc"
+            if not os.path.isfile(file_path):
+                raise Exception(
+                    "The Overcloud rc file does"
+                    " not exist {}".format(file_path))
+
+            with open(file_path) as rc_file:
+                for line in rc_file:
+                    if 'OS_USERNAME' in line:
+                        UC_USERNAME = line.split('OS_USERNAME=')[
+                            1].strip("\n").strip("'")
+                    elif 'OS_PASSWORD' in line:
+                        UC_PASSWORD = line.split('OS_PASSWORD=')[
+                            1].strip("\n").strip("'")
+                    elif 'OS_PROJECT_NAME' in line:
+                        UC_PROJECT_ID = line.split('OS_PROJECT_NAME=')[
+                            1].strip("\n").strip("'")
+                    elif 'OS_AUTH_URL' in line:
+                        UC_AUTH_URL = line.split('OS_AUTH_URL=')[
+                            1].strip("\n").strip("'")
+        except Exception as error:
+            message = "Exception {}: {}".format(
+                type(error).__name__, str(error))
+            logger.error("{}".format(message))
+            raise Exception(
+                "Failed to get overcloud details from"
+                " the overcloud rc file {}".format(file_path))
+
```

Comment:
credential helper class should provide what you need already

Reply:
Fixed